### PR TITLE
FreeRTOS-Plus: Unit testing Infrastructure and examples

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR intends to add the Unit testing infrastructure (CMock and Unity) to FreeRTOS-Plus directory for easier development of Unit tests. The main goal of this PR is to provide a base for Unit-testing the FreeRTOS-Plus-TCP source tree.
Additionally, some examples along with a Makefile are provided to help setup the infrastructure quickly. The README file is also written with the intent of making the setup as easy as possible.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Test/Unit-Tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+directories:
+	mkdir -p ${BUILD_DIR}
+	mkdir -p ${MOCKS_DIR}
+	mkdir -p ${LIB_DIR}
+	mkdir -p ${BIN_DIR}
+	mkdir -p ${GEN_DIR}
+
+mocks: directories
+	cd ${BUILD_DIR} && ruby  ${CMOCK_LIB_DIR}/cmock.rb -o${CONFIGURATION_DIR}/project.yml ${MOCK_HEADERS}
+	#@echo "\n"
```
Comment (alfred2g) [inline FreeRTOS-Plus/Test/Unit-Tests/Makefile:57]: commented

File: FreeRTOS-Plus/Test/Unit-Tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	@echo "Cleaned...\n"
+
+run:  ${BIN_DIR}/${EXECUTABLE}
+	${BIN_DIR}/${EXECUTABLE}
+
+coverage: run
+	lcov --base-directory . --directory . -c --rc lcov_branch_coverage=1 --rc genhtml_branch_coverage=1  -o build/cmock_test.info
+	genhtml build/cmock_test.info --branch-coverage --output-directory build/coverage_html
+
+#lcov --remove cmock_test.info "/usr*" --o cmock_test.info
```
Comment (alfred2g) [inline FreeRTOS-Plus/Test/Unit-Tests/Makefile:111]: commented

File: FreeRTOS-Plus/Test/Unit-Tests/Makefile (hunk 1 above)
Comment (AniruddhaKanhere) [reply FreeRTOS-Plus/Test/Unit-Tests/Makefile]: Removed. Thanks.

File: FreeRTOS-Plus/Test/Unit-Tests/Makefile (hunk 2 above)
Comment (AniruddhaKanhere) [reply FreeRTOS-Plus/Test/Unit-Tests/Makefile]: Removed. Thanks.

**Pull request merged**